Emby.Server.Implementations/Session/SessionManager.cs

```diff
@@ -721,6 +721,36 @@ public async Task OnPlaybackStart(PlaybackStartInfo info)
             {
                 foreach (var user in users)
                 {
+                    var activeStreams = Sessions
+                        .Where(i => i.UserId.Equals(user.Id)
+                            && i.DeviceId != session.DeviceId
+                            && i.NowPlayingItem is not null)
+                        .ToList();
+                    int maxActiveStreams = user.MaxActiveStreams;
+                    _logger.LogInformation("Current/Max streams for user {User}: {Streams}/{Max}", user.Username, activeStreams.Count(), maxActiveStreams);
+                    while (maxActiveStreams >= 1 && activeStreams.Count() >= maxActiveStreams)
+                    {
+                        try
+                        {
+                            var activeStream = activeStreams.First();
+                            await SendPlaystateCommand(
+                                activeStream.Id,
+                                activeStream.Id,
+                                new PlaystateRequest()
+                                {
+                                    Command = PlaystateCommand.Stop,
+                                    ControllingUserId = activeStream.UserId.ToString(),
+                                    SeekPositionTicks = activeStream.PlayState?.PositionTicks
+                                },
+                                CancellationToken.None).ConfigureAwait(true);
+                            activeStreams.RemoveAt(0);
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogDebug(ex, "Error calling SendPlaystateCommand for stopping max active session {Session}.", session.Id);
```

Comment:
Just realized this is wrong (copy/pasta got me here), for two reasons:
1.) This is using the sessionId of the new session instead of the one it's attempting stopping
2.) Two this will not work if I update it to use activeStream.Id, which is the correct one, as currently activeStream is currently declared inside the try-block, so I don't think the catch-block will be able to access it.

To capture this and have it flow correctly, I'd likely have to move the declaration of activeStream to right before the try-block and possibly move the RemoveAt() to after the end of catch-block.

Not completely sure about the latter, but I think from a while-looping perspective, this might create the possibility of an infinite loop as it might not be able to ever remove any items from activeStreams if the try-block fails before it gets to the RemoveAt().

Reply:
Fixed this and other minor issues.